packages/typespec-azure-resource-manager/src/namespace.ts

```diff
@@ -233,7 +233,7 @@ export const $armProviderNamespace: ArmProviderNamespaceDecorator = (
     }
 
     // Set route options for the whole namespace
-    setRouteOptionsForNamespace(program, entity, {
+    setRouteOptionsForNamespace(program, program.getGlobalNamespaceType(), {
```

Comment:
that most likely will cause issue with versioning but if it works(intresting hack) then probably just traversing up the entity namespace till the top will work

Reply:
it could work with setting to top level namespace.

Reply:
@timotheeguerin Would this change look good for you?